modules/saml/src/Controller/ServiceProvider.php

```diff
@@ -105,6 +105,46 @@ public function setAuthUtils(Utils\Auth $authUtils): void
     }
 
 
+    /**
+     * Start single sign-on for an SP identified with the specified entityID
+     *
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     * @param string $sourceId
+     * @return \SimpleSAML\HTTP\RunnableResponse
+     */
+    public function login(Request $request, string $sourceId): RunnableResponse
+    {
+        $as = new Auth\Simple($sourceId);
+        if (!($as->getAuthSource() instanceof SP)) {
+            throw new Error\Exception('Authsource must be of type saml:SP.');
+        }
+
+        if (!$request->query->has('ReturnTo')) {
+            throw new Error\BadRequest('Missing ReturnTo parameter.');
+        }
+        $returnTo = $request->query->get('ReturnTo');
+
+        /**
+         * Setting up the options for the requireAuth() call later..
+         */
+        $httpUtils = new Utils\HTTP();
+        $options = [
+            'ReturnTo' => $httpUtils->checkURLAllowed($returnTo),
+        ];
+
+        /**
+         * Allows a saml:idp query string parameter specify the IdP entity ID to be used
+         * as used by the DiscoJuice embedded client.
```

Comment:
DiscoJuice is long dead... so do we need this?

Reply:
The setting itself seems not specific to DiscoJuice.. It is referenced in `modules/saml/docs/sp.md`

Reply:
Maybe just drop it from the comment then... referencing the dead product will likely make the same question arise in the future?

Reply:
I think we can drop the entire code-block since the only references are to the SP authsource.